test/fuse_attention.cpp

```diff
@@ -38,11 +38,19 @@
 #include <reduce.hpp>
 #include <utility>
 
+MIGRAPHX_DECLARE_ENV_VAR(MIGRAPHX_ENABLE_FLASH_DECODING);
+
 static void run_pass(migraphx::program& p)
 {
     migraphx::run_passes(p, {migraphx::fuse_attention{}, migraphx::dead_code_elimination{}});
 }
 
+static void run_pass(migraphx::program& p, std::size_t num_splits)
```

Comment:
No need for an extra overload. I would just add the pass as a default parameter:

```cpp
static void run_pass(migraphx::program& p, migraphx::fuse_attention fa = {})
{
    migraphx::run_passes(p, {fa, migraphx::dead_code_elimination{}});
}
```

And then you can run this with `run_pass(p1, {.flash_decoding_num_splits=2})`.